lib/rspec/core/formatters/base_text_formatter.rb

```diff
@@ -63,11 +63,11 @@ def seed(notification)
         #
         # @param _notification [NullNotification] (Ignored)
         def close(_notification)
-          return if output.closed?
-
-          output.puts
-
-          output.flush
+          unless output.closed?
+            output.puts
+            output.flush
+          end
+          super
```

Comment:
There’s a check if output_supports_sync && !output.closed? in super. It feels right to keep the guard clause to only call super if output wasn’t closed.

Reply:
I figure each `close` should be responsible for its own guards which is why I'm unconditionally calling super.

Reply:
To make sure we’re talking about the same: have you seen the build failure caused by the `@old_sync` not being initialized?

I have no objections to separate guard clauses.
What do you think of splitting this statement in the base class? Will it work around the failure?
```diff
- output.sync = @old_sync if output_supports_sync && !output.closed?
+ return if output.closed?
+ output.sync = @old_sync if output_supports_sync
```

Reply:
I saw that failure, I'll dig into it a bit more on Monday but my guess is that it's actually a problem with the test rather than the implementation. `@old_sync` should always be set if `output_supports_sync` in the normal use case so I don't think it's a guard problem. Certainly seems to be true of ruby 3.1+, anyway.

Reply:
My wild guess is that `start` which calls[ start_sync_output does not init that instance var](https://github.com/rspec/rspec-core/blob/38abe6e0cc2c36ed7ebf9618072d66a8d5965191/lib/rspec/core/formatters/base_formatter.rb#L56) because output_supports_sync [doesn’t support sync](https://github.com/rspec/rspec-core/blob/38abe6e0cc2c36ed7ebf9618072d66a8d5965191/lib/rspec/core/formatters/base_formatter.rb#L64), at least in specs.

My guess is that 3.1 is smart enough not to warn when `@foo if false`, while earlier versions aren’t.

I’m eager to know what really causes such behavior.

Reply:
I can't recreate this error locally, so I tried to move the guard up to within `BaseFormatter#close`. If that fixes things, I can look into improving it somewhat.

Reply:
I'm also curious what rspecs support stance is, given that ruby [3.0 and older are all EOL](https://www.ruby-lang.org/en/downloads/branches/) and 3.1+ are passing with my changes.

Reply:
We support 1.8.7+, and plan on dropping everything below 2.5 in the next major version.

No worries, if you’re not up to add cross-version support, I have some spare time next week.

Reply:
This version of RSpec supports 1.8.7 to current, in order to be merged the build must pass